r/R/type.R

```diff
@@ -320,11 +320,27 @@ NestedType <- R6Class("NestedType", inherit = DataType)
 #' @return An Arrow type object inheriting from DataType.
 #' @export
 #' @seealso [dictionary()] for creating a dictionary (factor-like) type.
-#' @examplesIf arrow_available()
+#' @examplesIf arrow_available() && arrow_with_dataset()
 #' bool()
 #' struct(a = int32(), b = double())
 #' timestamp("ms", timezone = "CEST")
 #' time64("ns")
+#'
+#' # Use the cast method to change the type of data contained in Arrow Objects.
+#' # Please check the documentation of each data object class for details.
+#' Scalar$create(0L)$cast(int64())$cast(timestamp("ns"))
```

Comment:
Let's break this example up into two lines to make it easier to read - one for object creation and a separate operation for casting the object to a different type.

Please also just use a single cast in this.

Reply:
I wanted to include an example of a cast to timestamp type (since it takes a argument), but to cast to timestamp type we must first cast to int64.

Reply:
Just to check - what's the significance of the fact that it takes an argument?

Reply:
I was not aware that the type argument of the create method can be of type int64 from the beginning.
By setting it to type int64 from the beginning, cast is now executed only once.